src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -846,6 +862,15 @@ export class TerminalInstance implements ITerminalInstance {
 			this._messageTitleListener = null;
 		}
 	}
+
+	public eventuallyGetShellName(): void {
```

Comment:
Let's return the promise here. Also a better name would be `updateShellName` (or `updateWindowsShellName` if it's in `TerminalInstance`. Since `TerminalInstance.setTitle` is on the interface let's move this whole function into `WindowsShellService`.